### ISSUE TITLE
Service Forwarding IP

### ISSUE THREAD

**Issue by penguin2839** [F5Networks/f5-appsvcs-extension#458]
### **Enviornment**

- Application Services Version: 3.19.1
- BIG-IP Version: 15.1.0.4

### **Summary**

I am trying to create a service forwarding VIP but I keep getting this error: "/enterprise_networking/radius_coa_tis/radius_coa_tis_udp_1700: # is NOT valid"

### **Steps To Reproduce**

Post the following declaration to the BIG-IP assuming enterprise_networking is a valid tenant. 

```
{
  "class": "AS3",
  "action": "patch",
  "patchBody": [{
    "op": "add",
    "path": "/enterprise_networking/radius_coa_tis",
    "value": {
      "class": "Application",
      "template": "generic",
      "radius_coa_tis_udp_1700": {
        "class": "Service_Forwarding",
        "forwardingType": "ip",
        "layer4": "udp",
        "snat": {
          "use": "radius_coa_snat_pool"
        },
        "virtualAddresses": [
          ["0.0.0.0/0",
          "10.7.28.0/24"]
        ],
        "virtualPort": 1700,
        "pool": "",
        "persistenceMethods": []
      },
      "radius_coa_snat_pool": {
        "class": "SNAT_Pool",
        "snatAddresses": [
          "10.7.26.114"
        ]
      }
    }
  }]
}
```
### **Expected Behavior**

The declaration would post with no errors. 

### **Actual Behavior**

```
{
    "code": 422,
    "errors": [
        "/enterprise_networking/radius_coa_tis/radius_coa_tis_udp_1700: # is NOT valid"
    ],
    "declarationFullId": "",
    "message": "declaration is invalid"
}
```

**Comment by dstokesf5** [comment]
The `pool` and `persistenceMethods` properties are not supported on the `Service_Forwarding` class: https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/refguide/schema-reference.html#service-forwarding

The following PATCH body succeeds:
```json
{
  "class": "AS3",
  "action": "patch",
  "patchBody": [{
    "op": "add",
    "path": "/enterprise_networking/radius_coa_tis",
    "value": {
      "class": "Application",
      "template": "generic",
      "radius_coa_tis_udp_1700": {
        "class": "Service_Forwarding",
        "forwardingType": "ip",
        "layer4": "udp",
        "snat": {
          "use": "radius_coa_snat_pool"
        },
        "virtualAddresses": [
          ["0.0.0.0/0",
          "10.7.28.0/24"]
        ],
        "virtualPort": 1700
      },
      "radius_coa_snat_pool": {
        "class": "SNAT_Pool",
        "snatAddresses": [
          "10.7.26.114"
        ]
      }
    }
  }]
}
```

**Comment by melkypie** [comment]
If they are not supported should they not be taken out of documentation, since it clearly shows them being supported in schema reference? Also noticed that `virtualPort` is not included in `Service_Forwarding` which is a pretty critical option as it won't let you create a service without defining it and will keep throwing you this error: `"\\\"/TENANT/10.0.0.0:undefined\\\" contains invalid or ambiguous service undefined"`

**Comment by dstokesf5** [comment]
It turns out that the tooling we use to generate the schema documentation is not handling the Service_Forwarding class properly. It is including several unsupported properties and failing to document supported properties, such as virtualPort. I have added AUTOTOOL-1779 to our internal product backlog to find a way to get the tooling to generate more accurate documentation.

**Comment by sunitharonan** [comment]
If you are still looking for a fix, please reach out to us at automation_toolchain_pm@f5.com

**Comment by sunitharonan** [comment]
Closing as this has been released in AS3 3.48.0
